sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2098,6 +2098,12 @@ object SQLConf {
         "defined by `from` and `to`.")
       .booleanConf
       .createWithDefault(false)
+
+  val ADD_DIRECTORY_USING_RECURSIVE = buildConf("spark.sql.addDirectory.recursive")
+    .doc(" When true, users can add directory by passing path of a directory to ADD FILE" +
+      "command of SQL. If false, then only a single file can be added.")
+    .booleanConf
+    .createWithDefault(false)
```

Comment:
Does Hive have an option for this feature for turning off/on it?

Reply:
No. There is no such option in hive. It is enabled by default.

Reply:
If so, how about following that?; just adding an option under `spark.sql.legacy.xxx` then enabling it by default. Also, we need to update the migration guide, too.

Reply:
@maropu Do we need to add this under `spark.sql.legacy.xxx` ? Or is it fine this way only??
@HyukjinKwon @dongjoon-hyun